File: src/dotnet/Common/Models/Orchestration/LongRunningOperation.cs
using System.Text.Json.Serialization;
using FoundationaLLM.Common.Constants.Orchestration;

namespace FoundationaLLM.Common.Models.Orchestration
{
    /// <summary>
    /// Represents the current state of a long-running operation.
    /// </summary>
    public class LongRunningOperation
    {
        /// <summary>
        /// The identifier of the long-running operation.
        /// </summary>
        [JsonPropertyName("id")]
        public string? Id => OperationId;

        /// <summary>
        /// The document type.
        /// </summary>
        [JsonPropertyName("type")]
        public string Type => LongRunningOperationTypes.LongRunningOperation;

        /// <summary>
        /// The identifier of the long-running operation.
        /// </summary>
        [JsonPropertyName("operation_id")]
        public string? OperationId { get; set; }

        /// <summary>
        /// The status of the long-running operation.
        /// </summary>
        [JsonPropertyName("status")]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public required OperationStatus Status { get; set; }

        /// <summary>
        /// The message describing the current state of the operation.
        /// </summary>
        [JsonPropertyName("status_message")]
        public string? StatusMessage { get; set; }

        /// <summary>
        /// The time stamp of the last update to the operation.
        /// </summary>
        [JsonPropertyName("last_updated")]
        public DateTime? LastUpdated { get; set; } = DateTime.UtcNow;

        /// <summary>
        /// The time-to-live (TTL) for the log entry.
        /// </summary>
        [JsonPropertyName("ttl")]
        public int TTL { get; set; } = 604800;

        /// <summary>
        /// The result of the operation.
        /// </summary>
        [Newtonsoft.Json.JsonIgnore]
        [JsonPropertyName("result")]
        public object? Result { get; set; }
    }
}

File: src/python/PythonSDK/foundationallm/models/operations/long_running_operation.py
from datetime import datetime
from pydantic import BaseModel, Field
from typing import Optional
from foundationallm.models.orchestration import CompletionResponse

class LongRunningOperation(BaseModel):
    """
    Class representing a long running operation.
    """
    operation_id: str = Field(description='The unique identifier for the operation.')
    status: str = Field(description='The status of the operation.')
    status_message: Optional[str] = Field(description='The message associated with the operation status.')
    last_updated: Optional[datetime] = Field(default=None, description='The timestamp of the last update to the operation.')
    result: Optional[CompletionResponse] = Field(default=None, description='The result of the operation.')
